test/libsolidity/util/TestFunctionCall.cpp

```diff
@@ -154,40 +163,87 @@ string TestFunctionCall::formatBytesParameters(
 	if (_bytes.empty())
 		return {};
 
-	_errorReporter.warning("The call to \"" + _signature + "\" returned \n" + BytesUtils::formatRawBytes(_bytes));
+	if (_failure)
+	{
+		ParameterList defaultParameters;
+
+		defaultParameters.push_back(Parameter{bytes(), "", ABIType{ABIType::HexString, ABIType::AlignNone, 4}, FormatInfo{}});
+		defaultParameters.push_back(Parameter{bytes(), "", ABIType{ABIType::Hex}, FormatInfo{}});
+		defaultParameters.push_back(Parameter{bytes(), "", ABIType{ABIType::UnsignedDec}, FormatInfo{}});
+		defaultParameters.push_back(Parameter{bytes(), "", ABIType{ABIType::String}, FormatInfo{}});
 
-	boost::optional<ParameterList> abiParams = ContractABIUtils::parametersFromJson(
-		_errorReporter,
-		m_contractABI,
-		_signature
-	);
+		os << BytesUtils::formatBytesRange(_bytes, defaultParameters, _highlight);
```

Comment:
What happens if the data is not according to the given types above? This may well be the case later.

Reply:
The format would break. We might want to check if given bytes *can* be formatted before every call to `formatBytesRange`. But I would keep this in mind and implement it at a later point in time.

Reply:
I would prefer something that is working in all cases and just is not nice in some cases instead of it breaking in some cases.

Reply:
Wait, are you talking about the "special" case of a `FAILURE` or in general? I just checked and got a weird output in case the returned bytes do not match the given encoding:
```
function u() public returns (T[2] memory) {
    return [T(23, 42, "any"), T(555, 666, "any")];
}
// ----
// u() -> 0x20, 0x40, 0xE0, 23, 42, 0x60, 3, hex"ef0042", 555, 666, 0x60, 3
```
results in
```
// u() -> 0x20, 0x40, 0xE0, 23, 42, 0x60, 3, hex"616e79", 0, 9311354880, 0x29A000000, 1610612736, 56716921
```
which breaks the test after updating it. At least you have the actual returned data printed as a hint:
```
  Warning: The call to "u()" returned 
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,20]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,40]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,e0]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,17]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2a]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,60]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3]
  [61,6e,79,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,2b]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,2,9a]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,60]
  [0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,3]
  [61,6e,79,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0]
```

Fixing this might be not in the scope of this PR. Shall we create an issue?

Reply:
Yes, please create an issue.